yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -657,6 +657,9 @@ private[spark] class Client(
           case e: ApplicationNotFoundException =>
             logError(s"Application $appId not found.")
             return (YarnApplicationState.KILLED, FinalApplicationStatus.KILLED)
+          case ie: IOException =>
+            logError(s"Application $appId occurred an unexpected IOException.")
+            return (YarnApplicationState.KILLED, FinalApplicationStatus.KILLED)
```

Comment:
BTW, as a suggestion, I'd remove this exception handling from here, and wrap the `run` method in an exception handler instead.